### PR TITLE
feat: Support for TSVs, JSONL, and caching

### PR DESCRIPTION
**Pull request by sebastianswms**
Feature additions:

- Custom Singer stream name.
- CSV/TSV support
  - Automatically recognizing TSV vs. CSV files.
  - Custom delimiter and quoting character for CSVs/TSVs.
- File compression.
  - Automatically recognizing five encodings.
- Local caching
  - Developer-only setting allowing alternative options.
- JSONL files.
  - Multiple type coercion strategies
  - Scaffolding for future sampling strategies.

### CODE REVIEW

File: meltano.yml [hunk 1]
```diff
@@ -8,23 +8,30 @@ plugins:
   extractors:
   - name: tap-file
     namespace: tap_file
-    pip_url: -e .[s3]
+    pip_url: -e .
```
Comment (visch) [inline meltano.yml:11]: we don't want to install for s3 by default anymore?

File: tap_file/streams.py [hunk 2]
```diff
@@ -3,45 +3,151 @@
...
-class CSVStream(FileStream):
-    """Stream for reading CSVs."""
-
-    name = "CSV"
+class DelimitedStream(FileStream):
+    """Stream for reading CSVs and TSVs."""
 
     def get_rows(self) -> Generator[dict[str | Any, str | Any], None, None]:
-        """Retrive all rows from all CSVs.
+        """Retrive all rows from all *SVs.
```
Comment (visch) [inline tap_file/streams.py:18]: 😆

File: tap_file/streams.py [hunk 3]
```diff
@@ -3,45 +3,151 @@
...
+                    msg = (
+                        "Configuration option 'delimiter' is set to 'detect' but a "
+                        "non-csv non-tsv file is present. Please manually specify "
+                        "'delimiter'."
+                    )
+                    raise RuntimeError(msg)
+            else:
+                delimiter = self.config["delimiter"]
+
+            with self.filesystem.open(
```
Comment (visch) [inline tap_file/streams.py:64]: How does fsspec handle encodings? Does it default to utf8. We can make a seperate issue for this but I'm just remembering https://github.com/MeltanoLabs/tap-csv/blob/main/tap_csv/client.py#L117 and https://github.com/MeltanoLabs/tap-csv/pull/88

**Review by visch — APPROVED** [review]

File: meltano.yml (hunk 1 above)
Comment (sebastianswms) [reply meltano.yml:11]: Is that what we ended up deciding? I don't remember. When I saw that it was installed by default I thought it was an error that I'd forgotten to remove.

File: tap_file/streams.py (hunk 3 above)
Comment (sebastianswms) [reply tap_file/streams.py:64]: Yes, it defaults to UTF8. https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=encoding:#fsspec.gui.FileSelector.open_file

**Pull request merged**